src/runtime/pytuple.cs

```diff
@@ -74,7 +74,11 @@ private static IntPtr FromArray(PyObject[] items)
                 IntPtr ptr = items[i].obj;
                 Runtime.XIncref(ptr);
                 int res = Runtime.PyTuple_SetItem(val, i, ptr);
-                PythonException.ThrowIfIsNotZero(res, () => Runtime.Py_DecRef(val));
+                if (res == 0)
+                {
+                    Runtime.Py_DecRef(val);
+                    PythonException.ThrowIfIsNotZero(res);
```

Comment:
NIT: since you made corresponding check already, you can just `throw new PythonException()`